File: Day02/Medium/Solution.py
class Solution:
    def selectionSort(self, arr):
        n = len(arr)
        for i in range(n - 1):
            maxi = arr[0]
            maxiInd = 0
            for j in range(n - i):
                if arr[j] > maxi:
                    maxi = arr[j]
                    maxiInd = j
            arr[n - i - 1], arr[maxiInd] = arr[maxiInd], arr[n - i - 1]

if __name__ == "__main__":
    sol = Solution()
    n = int(input("Enter number of elements: "))
    arr = list(map(int, input("Enter elements: ").split()))

    sol.selectionSort(arr)

    print("Sorted array:", *arr)

File: Day02/Medium/Solution.java
import java.util.*;

class Solution {
    public void selectionSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            int maxi = arr[0];
            int maxiInd = 0;
            for (int j = 0; j < n - i; j++) {
                if (arr[j] > maxi) {
                    maxi = arr[j];
                    maxiInd = j;
                }
            }
            int temp = arr[n - i - 1];
            arr[n - i - 1] = arr[maxiInd];
            arr[maxiInd] = temp;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Solution sol = new Solution();

        System.out.print("Enter number of elements: ");
        int n = sc.nextInt();

        int[] arr = new int[n];
        System.out.print("Enter elements: ");
        for (int i = 0; i < n; i++) arr[i] = sc.nextInt();

        sol.selectionSort(arr);

        System.out.print("Sorted array: ");
        for (int x : arr) System.out.print(x + " ");
        System.out.println();
        sc.close();
    }
}
